src/main/java/com/hubspot/imap/protocol/message/ImapMessage.java

```diff
@@ -85,7 +86,14 @@ public ZonedDateTime getInternalDate() throws UnfetchedFieldException {
     }
 
     public Builder setInternalDate(String internalDate) {
-      this.internalDate = Optional.of(ZonedDateTime.parse(internalDate.trim(), INTERNALDATE_FORMATTER));
+      for (DateTimeFormatter formatter : INTERNALDATE_FORMATTERS) {
+        try {
+          this.internalDate = Optional.of(ZonedDateTime.parse(internalDate.trim(), formatter));
+          return this;
+        } catch (Exception e) {
```

Comment:
catch `DateFormatException` (or whatever it is) instead of blanket catch? NPE's scare me